evergreen/evergreen.yml

```diff
@@ -25,18 +33,66 @@ functions:
       binary: bash
       args:
       - "src/evergreen/test-centos7-evergreen.sh"
+  "archive":
+    command: subprocess.exec
+    params:
+      binary: bash
+      args:
+      - "src/evergreen/archive-evergreen.sh"
+  # Upload archives to the S3 directory for this build.
+  "s3 upload archive":
+    command: s3.put
+    params:
+      aws_key: ${release_aws_key}
+      aws_secret: ${release_aws_secret}
+      bucket: mongodb-mms-build-envoy-serverless
+      local_files_include_filter: ["*.tar.gz"]
+      local_files_include_filter_prefix: "archive"
+      remote_file: build/${build_id}/
+      permissions: public-read
+      content_type: "application/x-gzip"
+  "create push credentials":
+    command: shell.exec
+    params:
+      working_dir: .
+      script: |
+        set -e
+        # Create a config file for AWS S3 CLI
+        echo "creating config file"
+        echo "[default]" > conf.aws
+        chmod 700 conf.aws
+        echo aws_access_key_id=${release_aws_key} >> conf.aws
+        echo aws_secret_access_key=${release_aws_secret} >> conf.aws
+        # Create a credentials file for AWS S3 CLI
+        echo "creating credentials file"
+        echo "[default]" > cred.aws
+        chmod 700 cred.aws
+        echo region=us-east-1 >> cred.aws
+  "s3 push archive":
+    - command: shell.exec
+      params:
+        working_dir: .
+        script: |
+          set -evx
+          # Export the config file location
+          export CONFIG_DIR=`pwd`
+          export AWS_CONFIG_FILE=$CONFIG_DIR/conf.aws
+          export AWS_CREDENTIAL_FILE=$CONFIG_DIR/cred.aws
+          export BUCKET=mongodb-mms-build-envoy-serverless
+          # Copy binary archives from build directory to tarballs directory
+          aws s3 ls s3://$BUCKET/build/${build_id}/  \
+            | awk '{print $NF}'  \
+            | grep "envoy-serverless-.*\.tar\.gz"  \
+            | xargs -I{} -n1 aws s3 cp s3://$BUCKET/build/${build_id}/{} s3://$BUCKET/tarballs/
 
 buildvariants:
 - name: centos7
   display_name: CentOS 7
   run_on:
     # Tasks run in x86 CentOS7 containers, so the Linux distro doesn't have to match.
     - rhel80-xlarge
-  expansions:
-    release: 7
-    image: centos:7
-    arch: x86_64
   tasks:
```

Comment:
Should we run `compile` and `test` tasks on PR merge, and the `push` task on merge to master? Looks like we could use something like `patchable: false` to configure `push` to only run on merge: https://github.com/evergreen-ci/evergreen/wiki/Project-Configuration-Files#limiting-when-a-task-will-run

Reply:
Running patches on PRs needs more config on github. Since we won't develop in this repo a lot, I didn't set it.

Great point on `patchable: false` though. I needed patch builds to test my changes, but we should set it. Done.

Reply:
Seems like you can just click this button in the evergreen project settings ([source](https://github.com/evergreen-ci/evergreen/wiki/Using-Repo-Level-Settings#how-it-works))
![Screen Shot 2023-01-04 at 9 17 43 AM](https://user-images.githubusercontent.com/7087489/210574901-739fb31a-7a73-4132-9319-0d150c844949.png)
> Running patches on PRs needs more config on github.

Reply:
Thanks! That's great to know. Will do.